### ISSUE TITLE
GG Max: Placed objects permanently disappearing upon concurrent saves/tests

### ISSUE THREAD

**Issue by bohemivn** [TheGameCreators/GameGuruRepo#4775]
Hello there, reposting a discussion of the issue I had with synchromesh over on the steam forums today.

https://steamcommunity.com/app/1247290/discussions/0/5956623715026411763/?tscn=1689454676

**dotcomdot 17 hours ago**

**Half of the objects I place disappear permanently**
_If I save or go to test the level to make sure they haven't broken anything. They just vanish, never to be seen again. I just placed down about 10 of the same small object, and after saving only 2 are left in the area I put them all._

_This is pretty much program breaking considering I cant trust that anything I place will actually stay there as I continue working on a project._

-----
**synchromesh 11 hours ago**

_Hi there Could you try the Experimental build and see if this does the same thing for you ?_

Right click on GameGuruMax in your Steam Library
Select Properties
Select Beta
Enter this code
lettesterin456

Let it update
Reboot and try again

Or you could try verifying the installation via steam for any corrupt files_

-----
**dotcomdot 16 minutes ago**

_Hey there, synchromesh._

_Thanks for the response. I wanted to make sure to give some multifaceted feedback, so I verified my game files first, and tested. I gradually added objects, saving after every few. Finally I reached a save where it wiped what I had just placed and stuff I had placed and saved separately prior in the same session._

_Testing the experimental beta- There was an error banner I'd not seen before upon first load of the level that mentioned behaviors.. not being able to to be found I believe? pardon my poor memory. I followed near the same objects placements as the test after verifying integrity. Placed the first batch, saved, and they all disappeared again, while showing that same error at the climax of the save._

_Upon another re-placement of what disappeared again, and further placements I was yet to recreate the disappear a second time on the branch you provided- I thought maybe something in the backend had fixed whatever was causing it- until I did the ultimate test of a full program save/quit._

_I relaunched gameguru and loaded my level and was met with a banner saying "some behaviors have been replaced with newer workshop content" and everything I had placed in the beta session was wiped._

**Comment by davetgc** [comment]
Hi,

Could you please supply the FPM and the LOG file if there is one for the level where this is happening? You can find both in your Mapbank folder.

**Comment by davetgc** [comment]
Thanks.

**Comment by bohemivn** [comment]
Granted!

Sorry about the delay

**Comment by LeeBamberTGC** [comment]
@bohemivn Thanks for the FPM file you sent. In the next EXP build, I noticed some media it links to that I do not have in my test area, so if you can zip up and send me anything that is not available as a DLC that would be great (i.,e. Buildings you made)

As to the missing disappearing objects, I acceted the missing media, and added 7 red barrels, saved and reloaded and they stayed in the level fine:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/39a4e115-cb12-447f-a5d4-d1509e321055)

Finally, do not worry about the workshop warning, just ensure you are logged into the Steam Client to get the latest version of those scripts you depend on.

**Comment by bohemivn** [comment]
[buildingeditor.zip](https://github.com/TheGameCreators/GameGuruRepo/files/12088436/buildingeditor.zip)
[other (2).zip](https://github.com/TheGameCreators/GameGuruRepo/files/12088437/other.2.zip)

oh yeah! look at that. Sorry, the map uses a lot of buildings I made within the editor. 

Fortunately, I haven't gone as far as to really add any assets not from within the program/DLC's aside from I think just a volleyball net asset? I've added a second zip with that and some fpe's of other group structure things I created/had saved in folders for the level

**Comment by smallg2020** [comment]
have you checked the "detailed object list" to see if the objects are being deleted / removed or if they are being hidden / moved?
it's an option in the drop down bar next to where you "add" new objects into your level on the left there. (there's options for how to order your listed objects -> if the option isn't there you will need to enable it in your settings), this will let you select individual versions of each object and see which ones are in the map etc.

**Comment by LeeBamberTGC** [comment]
Will close this for the moment, the latest EXP build has some fixes for level corruptions so will reopen if can reproduce with the new build, thanks for the report!

**Comment by bohemivn** [comment]
Hey Lee! I've been busy with work lately, and today I finally had time to try and see if your team's work fixed this issue. I thought it was fixed until I started going heavy on some item placement and saved, and lo and behold I was able to recreate the issue again. I'm going to try and record as short of a video as I can to capture what is happening with me speaking light commentary and upload it somewhere so that you can view it, since I'd assume it'd be too large to place directly here.

**Comment by LeeBamberTGC** [comment]
@bohemivn Another report had something similar, but when I got the level and could not load two smart objects, the issue went away and was able to work for 10 minutes with no further object disappearances. What I ideally need is the level, any assets it uses, and then a video showing loading that level, editing the level and save/loading until the video shows a disappeared object. From this I can reproduce, locate and fix.

**Comment by bohemivn** [comment]
alright! here's a video. Everything file wise should be the same as when I last provided them to you, since this was my first attempt since your team's tinkering related to this ticket. Here's a video I tried to make as short as possible showing the dissapearing items. One thing to note: it is not native to the inside of the structure I used as an example. I have areas across the entire level that will sometimes lose objects due to this saving delete. 

I can happily upload this somewhere else if need be! Thanks for your ongoing help. One day this game will be alive.

https://drive.google.com/file/d/1Rjs63uXqlegvB9nt5xuvsIcAkCHY5anE/view?usp=share_link
